### PR TITLE
Improve dev tools to update browser variables list.

### LINKED ISSUE #164: Found 17 global variables with my module.
Hello. I want to use your package for my one. Not sure if that is not exist already. But anyway.
The problem that when I start to test [my package](https://github.com/shapkarin/global-diff) I found 17 global vars in the chrome incognito mode.
Here is the list:
```js
0: "onformdata"
1: "onwebkitanimationend"
2: "onwebkitanimationiteration"
3: "onwebkitanimationstart"
4: "onwebkittransitionend"
5: "onselectstart"
6: "onselectionchange"
7: "captureEvents"
8: "releaseEvents"
9: "webkitRequestAnimationFrame"
10: "webkitCancelAnimationFrame"
11: "webkitStorageInfo"
12: "chrome"
13: "onpointerrawupdate"
14: "webkitRequestFileSystem"
15: "webkitResolveLocalFileSystemURL"
16: "trustedTypes"
```
I forked your repo and want to add them if they all are the default now for Chrome too.

### LINKED ISSUE #166: Fix get-browser-globals.js and add additional update scripts for browser's environment.
I think that yours `get-browser-globals.js` don't works well test. Here what I got:
- run script from `get-browser-globals.js`
- copy my code to the json file
- run `npm run  test`
- get an error that `upperCase` letter must be `before LowerCased`

So I also fix the get-browser-globals.js script to avoid that.

### PR DESCRIPTION
**Pull request by shapkarin**
resolves #164 and also closes #166

### CODE REVIEW

File: globals.json [hunk 1]
```diff
@@ -227,20 +227,24 @@
 	"browser": {
 		"AbortController": false,
 		"AbortSignal": false,
-		"addEventListener": false,
```
Comment (shapkarin) [inline globals.json:230]: not sure why it's removed

**Comment by shapkarin** [comment]
That's what I trying to do:
- I cloned jshint and eslint and added it as `devDependencies`
- next I plan to add it to the `get-browser-globals.js` script
- in this way I can get the variables directly from the files
- not sure if they will be imported as well to the result module but we can try some build trick to make that works out

File: globals.json [hunk 2]
```diff
@@ -453,327 +347,69 @@
...
-		"RTCIceTransport": false,
-		"RTCPeerConnection": false,
-		"RTCPeerConnectionIceEvent": false,
-		"RTCRtpContributingSource": false,
-		"RTCRtpReceiver": false,
-		"RTCRtpSender": false,
-		"RTCSctpTransport": false,
-		"RTCSessionDescription": false,
-		"RTCStatsReport": false,
-		"RTCTrackEvent": false,
```
Comment (shapkarin) [inline globals.json:732]: @sindresorhus  something is missing. I guess we need to combine that `jshint` import method with the previous one somehow.

**Comment by shapkarin** [comment]
@sindresorhus 
I start to [automate](https://github.com/sindresorhus/globals/pull/165/commits/aad6d7a8e565d22229c68999ca15b86d0343120e) that somehow. At least for some steps from update's `readme.md`

### Usage
- Run `npm run get-jshint-browser`.
- Open an Incognito window in Chrome Canary and paste the above into the console.
- You'll now have a new object in your clipboard.
- Paste yours current clipboard to the field `my_browser` field at the root of the project.
- run `npm run combine-browser`
- Copy and paste the result from the `browser_vars.json` to the field `browser` at the `globals.json`

File: update/get-browser-globals/readme.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+### Usage
+- Run `npm run get-jshint-browser'.
```
Comment (shapkarin) [inline update/get-browser-globals/readme.md:2]:
```suggestion
- Run `npm run get-jshint-browser`.
```

**Comment by shapkarin** [comment]
@sindresorhus I hope it's a good one contribution and that can help me to make my package better.
Maybe even the `eslint` as well. I will be wait the review and some response.

**Comment by sindresorhus** [comment]
Can you describe how your PR is solving those referenced issues?

**Comment by sindresorhus** [comment]
We definitely don't want to fetch globals directly from JSHint. That readme reference just points out where the global originally came from.

**Comment by sindresorhus** [comment]
And there is already a similar PR open: https://github.com/sindresorhus/globals/pull/132 How is yours different?

**Comment by shapkarin** [comment]
This PR does not contains global browser variables update tool set
refactoring. I added the tools that help to extract that list from jshint,
combine it with yours approach to get them from console log and reface it a
little to avoid the test problems.

вс, 12 июля 2020 г. в 16:55, Sindre Sorhus <notifications@github.com>:

**Comment by shapkarin** [comment]
Currently my toolset is WIP. Not ready but almost, only after the PR I
realised that. You can look at it also at that repo for more recent version
as I remember, at least for updated readme
https://github.com/shapkarin/global-diff/tree/combine/update/get-browser-globals

вс, 12 июля 2020 г. в 17:02, Юрий Шапкарин <yu.shapkarin@gmail.com>:

**Comment by shapkarin** [comment]
> We definitely don't want to fetch globals directly from JSHint. That
readme reference just points out where the global originally came from.

JsHint is added as a devDependency so it’s not for the bundle. But for the
usage to the update the browsers var. As I remember I done that somehow
with it and updated list comes in the automated way. Not so wip as I
remember  I’m not with PC now, can check later.

Currently my toolset is WIP. Not ready but almost, only after the PR I

**Comment by shapkarin** [comment]
Sorry, messages are fragmented from the email answer. So I remember that -
I update the list with jshint + yours approach. And I save tools that I
used for this. And this or contains some version.

пн, 13 июля 2020 г. в 04:20, Юрий Шапкарин <yu.shapkarin@gmail.com>:

**Comment by shapkarin** [comment]
@sindresorhus At that PR
https://github.com/sindresorhus/globals/pull/132/files
I don’t see any see any automated script that get jshint vars that pass
your lint (test). and But, yes, I can change the devDepenecy from my repo
to some other.

пн, 13 июля 2020 г. в 04:25, Юрий Шапкарин <yu.shapkarin@gmail.com>:

**Comment by shapkarin** [comment]
@sindresorhus It’s the way to get jshint vars and from Canary or other
manual, combine them and pass the tests.

пн, 13 июля 2020 г. в 04:36, Юрий Шапкарин <yu.shapkarin@gmail.com>:

**Comment by shapkarin** [comment]
BTW thanks for your package and it’s basic update script, because it’s
inspire me to improve that. And that’s why I add that to you to your repo
As a PR and just an open an issue at the jshint’s one.


@sindresorhus It’s the way to get jshint vars and from Canary or other

**Comment by shapkarin** [comment]
@sindresorhus summary:
- jshint fork repo was imported as a devDependencies.
- this PR contains some Improve for dev tools to update the browser variables list. Not sure that it's not WIP.
- the same approach can be used for other environments.
- this PR contains browser variables list.

**Comment by sindresorhus** [comment]
Again, we don't want to fetch from JSHint. The globals there are outdated. The whole point of this repo is to be the source of truth.

**Comment by shapkarin** [comment]
OK, I got that. Maybe the reason that they want to keep all possible legacy support, not sure. I will close PR or later find some time to change the PR. Also it will be nice to change the `README.md` because it has some difference with yours response and create a misunderstanding.

**Comment by sindresorhus** [comment]
I would be happy to make the readme clearer, but I don't know what part is unclear.

**Comment by shapkarin** [comment]
> Extracted from JSHint and ESLint and merged.

**Comment by shapkarin** [comment]
I can try to make it more clear maybe too. When I will back to my PR. Also plan to compare it with the other one.

**Comment by shapkarin** [comment]
> Extracted from JSHint and ESLint and merged.

I decide that it's a normal process – extract it every time when you need to update + yours tool for browser vars. 

Also please check [that file](https://github.com/sindresorhus/globals/pull/165/files#diff-084045fe850b123ee962ff5af861219a) without that change I got an error at tests.

**Comment by sindresorhus** [comment]
Closing for lack of activity and also because this is not the solution we wanted.

**Pull request closed without merging**